sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataReaderDAO.java

```diff
@@ -887,6 +895,12 @@ private Record emptyRecord(Key key) {
                 Iterators.<RecordEntryRawMetadata>emptyIterator());
     }
 
+    @VisibleForTesting
+    public static Iterable<Row> getFilteredRows(Iterable<Row> rows, DateTime cutoffTime) {
+        UUID cutoffTimeUUID = TimeUUIDs.uuidForTimestamp(cutoffTime.toDate());
+        return () -> Iterators.filter(rows.iterator(), row -> (TimeUUIDs.compare(row.getUUID(CHANGE_ID_RESULT_SET_COLUMN), cutoffTimeUUID) < 0));
```

Comment:
I wonder if this would be slightly more performant if instead of constructing `cutoffTimeUUID` we extracted the time from the UUID using `TimeUUIDs.getTimeMillis()` and compared timestamps.  The effect is likely nominal, the only real change is that we wouldn't have to construct a new UUID object for every row, but since this would happen during Stash which reads every single row from every table even a slight performance change could have positive results.